compose/config/validation.py

```diff
@@ -269,10 +269,13 @@ def format_error_message(error, service_name):
     return '\n'.join(format_error_message(error, service_name) for error in errors)
 
 
-def validate_against_fields_schema(config, filename):
+def validate_against_fields_schema(config, filename, version=None):
+    schema_filename = "fields_schema.json"
+    if version:
+        schema_filename = "fields_schema_v{0}.json".format(version)
```

Comment:
I think with the suggested change to making a default version of 1 we can drop the `if` here, and make `version` a required position arg.